### PR TITLE
updated MenuItem to support description

### PR DESCRIPTION
**Pull request by laraharrow**
### :sparkles: Changes

- updated MenuItem to support description

### :white_check_mark: Requirements

- [ ] Includes test coverage for all changes
- [ ] Build and tests are passing
- [ ] Update documentation
- [ ] Updated CHANGELOG
- [ ] Checked for i18n impacts
- [ ] Checked for a11y impacts
- [ ] PR is ideally < ~400LOC

### :camera: Screenshots

Screen Shot 2020-08-10 at 3 40 36 PM

### CODE REVIEW

File: packages/components/src/Menu/MenuItem.tsx [hunk 1]
```diff
@@ -167,14 +173,23 @@ const MenuItemInternal: FC<MenuItemProps> = (props) => {
...
         {renderedIcon}
-        <span>{children}</span>
+        <Title>
+          {children}
+          {description && (
+            <Paragraph fontSize="xsmall" color="text2">
+              {description}
+            </Paragraph>
+          )}
+        </Title>
```
Comment (mdodgelooker) [inline packages/components/src/Menu/MenuItem.tsx:183]: If you change the `height` property to `min-height` on [ln 79 in `MenuItemLayout.tsx`](https://github.com/looker-open-source/components/blob/lara/MenuItem-description/comp1093/packages/components/src/Menu/MenuItemLayout.tsx#L79), then you can revert back to `span` here as well as the 2 other suggestions below...
```suggestion
        <span>
          {children}
          {description && (
            <Paragraph fontSize="xsmall" color="text2">
              {description}
            </Paragraph>
          )}
        </span>
```

File: packages/components/src/Menu/MenuItem.tsx [hunk 2]
```diff
@@ -167,14 +173,23 @@ const MenuItemInternal: FC<MenuItemProps> = (props) => {
...
         {detail && <Detail>{detail}</Detail>}
       </Component>
     </MenuItemLayout>
   )
 }
 
-export const MenuItem = styled(MenuItemInternal)``
+const Title = styled.span`
+  margin-top: auto;
+`
```
Comment (mdodgelooker) [inline packages/components/src/Menu/MenuItem.tsx:192]: ...and then you can remove this (see comment above)...
```suggestion
```

File: packages/components/src/Menu/MenuItem.tsx [hunk 3]
```diff
@@ -183,6 +198,15 @@ const Detail = styled.div`
...
 `
 
+export const MenuItem = styled(MenuItemInternal)`
+  display: flex;
+  ${({ description, theme: { space } }) =>
+    description &&
+    `
+      padding-bottom: ${space.small};
+    `}
+`
```
Comment (mdodgelooker) [inline packages/components/src/Menu/MenuItem.tsx:208]: ...and revert this as well (see comments above).
```suggestion
export const MenuItem = styled(MenuItemInternal)``
```

**Comment by jhardy** [comment]
@laraharrow can we add some margin to the top of the description, `xxsmall` on the top to give a bit of spacing between the description and the label. 

One thing I noticed as well is that the focus indicator does not play well with the descripton

image

**Comment by laraharrow** [comment]
@jhardy I fix the margin but I don't see the focus issue
Screen Shot 2020-08-12 at 10 27 01 AM

File: packages/components/src/Menu/MenuItem.tsx [hunk 4]
```diff
@@ -167,22 +173,34 @@ const MenuItemInternal: FC<MenuItemProps> = (props) => {
     >
       <Component href={href} role="menuitem" target={target}>
         {renderedIcon}
-        <span>{children}</span>
+        <span>
+          {children}
+          {description && (
+            <Paragraph fontSize="xsmall" color="text2">
```
Comment (mdodgelooker) [inline packages/components/src/Menu/MenuItem.tsx:179]: I think you can just put the margin here:
```suggestion
            <Paragraph fontSize="xsmall" mt="xxsmall" color="text2">
```

**Pull request merged**